packages/common/src/store/purchase-content/slice.ts

```diff
@@ -0,0 +1,80 @@
+import { Action, createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { ID } from 'models/Identifiers'
+
+import { ContentType, PurchaseContentStage } from './types'
+
+type OnSuccess = {
+  action?: Action
+  message?: string
+}
+
+type PurchaseContentState = {
+  stage: PurchaseContentStage
+  contentType: ContentType
+  contentId: ID
+  error?: Error
+  onSuccess?: OnSuccess
+}
+
+const initialState: PurchaseContentState = {
+  contentType: ContentType.TRACK,
+  contentId: -1,
+  error: undefined,
+  stage: PurchaseContentStage.START
+}
+
+const slice = createSlice({
+  name: 'purchase-content',
+  initialState,
+  reducers: {
+    startPurchaseContentFlow: (
+      state,
+      action: PayloadAction<{
+        contentId: ID
+        contentType?: ContentType
+        onSuccess?: OnSuccess
+      }>
+    ) => {
+      state.stage = PurchaseContentStage.START
+      state.error = undefined
+      state.contentId = action.payload.contentId
+      state.contentType = action.payload.contentType || ContentType.TRACK
+      state.onSuccess = action.payload.onSuccess
+    },
+    onBuyUSDC: (state) => {
```

Comment:
nit: the `on` makes these actions sound like callbacks to me, though I see why you did it like that

Reply:
I think this was following a pattern from elsewhere? I don't have strong opinions on how we do this, was just trying to make it consistent 😅